platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfigurationContainer.java

```diff
@@ -356,18 +356,16 @@ public Configuration maybeCreateDependencyScopeLocked(String name) {
 
     @Override
     public Configuration maybeCreateDependencyScopeLocked(String name, boolean verifyPrexisting) {
-        return doMaybeCreateLocked(new NoContextRoleBasedConfigurationCreationRequest(name, ConfigurationRoles.DEPENDENCY_SCOPE, problemsService), verifyPrexisting);
+        return doMaybeCreateLocked(new DefaultRoleBasedConfigurationCreationRequest(name, ConfigurationRoles.DEPENDENCY_SCOPE, problemsService), verifyPrexisting);
     }
 
     @Override
     public Configuration maybeCreateMigratingLocked(String name, ConfigurationRole role) {
-        AbstractRoleBasedConfigurationCreationRequest request = new NoContextRoleBasedConfigurationCreationRequest(name, role, problemsService);
+        DefaultRoleBasedConfigurationCreationRequest request = new DefaultRoleBasedConfigurationCreationRequest(name, role, problemsService);
```

Comment:
I think most of these `maybeCreate*` methods can now be migrated to the non-maybe versions, as these now all unconditionally fail if a configuration with the given name already exists